bctw-api/src/apis/deployment_api.ts

```diff
@@ -0,0 +1,98 @@
+import { Request, Response } from 'express';
+import { pgPool } from '../database/pg';
+import { getRowResults } from '../database/query';
+import { ICollar } from '../types/collar';
+import { getUserIdentifier } from '../database/requests';
+import { apiError } from '../utils/error';
+
+interface IDeployDevice extends ICollar {
+  critter_id: string;
+  data_start: Date;
+  data_end: Date;
+}
+
+/**
+ * Provides a mechanism for importing and linking telemetry devices and critters to a user.
+ * Aims to avoid collisions on any existing critters or devices by checking existing records.
+ *
+ * @param {IDeployDevice} data
+ * @param {string} user
+ */
+const deployDeviceDb = async (data: IDeployDevice, user: string) => {
+  const client = await pgPool.connect(); //Using client directly here, since we want this entire procedure to be wrapped in a transaction.
```

Comment:
Theres a helper function called transactionify that will wrap the SQL in a transaction for you.